Implementation/Services/index.md

```diff
@@ -0,0 +1,510 @@
+                var contentCache = umbracoContextReference.UmbracoContext.ContentCache;
+                //query the published content cache
+                IEnumerable<IPublishedContent> rootNodes = contentCache.GetAtRoot();
+                //retrieve a particular item with id 1234 from the published content cache
+                IPublishedContent particularItem = contentCache.GetById(1234);
+            }
+        }
+
+        public void Terminate()
+        {
+         
+        }
+    }
+}
+```
+
+#### Accesing Published Content Cache from a Content Finder / UrlProvider
+
+Inside a ContentFinder access to the content cache is provided via the PublishedRequest object:
+```csharp
+  public bool TryFindContent(PublishedRequest frequest)
+        {
+            var someContent = frequest.UmbracoContext.ContentCache.GetById(1234);
+```
+
+and insde a UrlProvider the GetUrl method has the current UmbracoContext injected:
+```csharp
+   public override UrlInfo GetUrl(UmbracoContext umbracoContext, IPublishedContent content, UrlProviderMode mode, string culture, Uri current)
+        {
+        var someContent = umbracoContext.ContentCache.GetById(1234);
+```
+
+## Custom Services and Helpers
+
+When implementing an Umbraco site, it is likely to have to execute similar code that accesses or operates on Umbraco data, in multiple places, perhaps using the Core Management Services or Umbraco Helpers.
+For example; Getting a list of the latest News Articles, or building a link to the site's News Section or Contact Us page. It's easy to repeat this kind of logic in multiple places, Views, Partial Views / Controllers etc, which is fine, but it's generally considered good practice to consolodate this logic into a single place.
+
+### Extension methods
+
+One option is to add 'Extension Methods' to the `UmbracoHelper` class.
+```csharp
+using Umbraco.Core.Models.PublishedContent;
+using Umbraco.Web;
+
+namespace Umbraco8.Extensions
+{
+    public static class UmbracoHelperExtensions
+    {
+            //NewsCategories //NewsSection
+            public static IPublishedContent GetNewsSection(this UmbracoHelper umbracoHelper)
+            {
+            return umbracoHelper.ContentSingleAtXPath("root/homePage/newsSection");
+            }
+     }
+}
+```
+
+anywhere there is reference to the UmbracoHelper, and a reference is added to the namespace the extension belongs to, it is possible to call the method by writing `Umbraco.GetNewsSection()`
+
+### Custom Services and Helpers
+
+Another option, is to make use of the underlying DI framework, and create custom Services and Helpers, that in turn can have the 'core' Management Services and Umbraco Helpers injected into them - (the same rules about transient services apply!). This approach enables the grouping together of similar methods within a suitably named service, and promotes the possibility of testing this custom logic outside of controllers and views.
```

Comment:
Not sure if this 'transient' rule makes sense ... see my notes above.

Reply:
No it doesn't, I have had another attempt to make it make clearer in sense! having the Lifespan info on the same page, makes it clearer too